doc/source/serve/tutorials/java.md

```diff
@@ -0,0 +1,145 @@
+:language: java
+:start-after: docs-strategy-start
+```
+
+This `Strategy` class is used to calculate the indicators of a number of banks.
+
+* The `calc` method is the entry of the calculation. The input parameters are the time interval of calculation and the map of the banks and their indicators. As we can see, the `calc` method contains a two-tier `for` loop, traversing each indicator list of each bank, and calling the `calcBankIndicators` method to calculate the indicators of the specified bank.
+
+- There is another layer of `for` loop in the `calcBankIndicators` method, which traverses each indicator, and then calls the `calcIndicator` method to calculate the specific indicator of the bank.
+- The `calcIndicator` method is a specific calculation logic based on the bank, the specified time interval and the indicator.
+
+This is the code that uses the `Strategy` class:
+
+```{literalinclude} ../../../../java/serve/src/test/java/io/ray/serve/docdemo/StrategyCalc.java
+:end-before: docs-strategy-calc-end
+:language: java
+:start-after: docs-strategy-calc-start
+```
+
+When the scale of banks and indicators expands, the three-tier `for` loop will slow down the calculation. Even if the thread pool is used to calculate each indicator in parallel, we may encounter a single machine performance bottleneck. Moreover, this `Strategy`  object cannot be reused as a resident service.
+
+## Converting to a Ray Serve Deployment
+
+Through Ray Serve, the core computing logic of `Strategy` can be deployed as a scalable distributed computing service.
+
+First, we can extract the indicator calculation of each institution into a separate `StrategyOnRayServe` class:
+
+```{literalinclude} ../../../../java/serve/src/test/java/io/ray/serve/docdemo/StrategyOnRayServe.java
+:end-before: docs-strategy-end
+:language: java
+:start-after: docs-strategy-start
+```
+
+Next, we start the Ray Serve runtime and deploy `StrategyOnRayServe` as a deployment.
+
+```{literalinclude} ../../../../java/serve/src/test/java/io/ray/serve/docdemo/StrategyCalcOnRayServe.java
+:end-before: docs-deploy-end
+:language: java
+:start-after: docs-deploy-start
+```
+
+The `Deployment.create` makes a `Deployment` object named "strategy". After executing `Deployment.deploy`, this "strategy" deployment is deployed in the instance of Ray Serve with 4 replicas, and we can access it for distributed parallel computing.
+
+## Testing the Ray Serve Deployment
+
+Now we can test the "strategy" deployment using RayServeHandle inside Ray:
+
+```{literalinclude} ../../../../java/serve/src/test/java/io/ray/serve/docdemo/StrategyCalcOnRayServe.java
+:end-before: docs-calc-end
+:language: java
+:start-after: docs-calc-start
+```
+
+At present, the calculation of each bank's each indicator is still in series, and just sended to Ray for execution. We can make the calculation concurrent, which not only improves the calculation efficiency, but also solves the bottleneck of single machine.
+
```

Comment:
what does "is still in series," mean? Might want to clarify. Do you mean serial?

Reply:
just sended to Ray --> sent to Ray